opta/commands/deploy.py

```diff
@@ -129,22 +131,26 @@ def deploy(
         outputs = Terraform.get_outputs(layer)
     except MissingState:
         outputs = {}
-    if "docker_repo_url" not in outputs or outputs["docker_repo_url"] == "":
-        logger.info(
-            "Did not find docker repository in state, so applying once to create it before deployment"
-        )
-        _apply(
-            config=config,
-            env=env,
-            refresh=False,
-            image_tag=None,
-            test=False,
-            local=local,
-            auto_approve=auto_approve,
-            stdout_logs=False,
-            detailed_plan=detailed_plan,
-        )
-    image_digest, image_tag = _push(image=image, config=config, env=env, tag=tag)
+
+    image_digest, image_tag = (None, None)
```

Comment:
something seems off with `image_tag` it's always None when calling apply?
can you double check?
and cover by unit test if indeed there was an error.

if we don't need it then we don't need this var

Reply:
Right now, every time the push occurs, we are passing in the image digest which is different every time.

Reply:
cool, something we will revisit later